massa-models/src/block.rs

```diff
@@ -759,28 +835,54 @@ impl Deserializer<BlockHeader> for BlockHeaderDeserializer {
             parents[slot.thread as usize],
         ));
 
-        let (rest, endorsements) = context(
+        let (rest, endorsements): (&[u8], Vec<Wrapped<Endorsement, EndorsementId>>) = context(
             "Failed endorsements deserialization",
             length_count::<&[u8], Wrapped<Endorsement, EndorsementId>, u32, E, _, _>(
                 context("Failed length deserialization", |input| {
                     self.length_endorsements_deserializer.deserialize(input)
                 }),
                 context("Failed endorsement deserialization", |input| {
-                    endorsement_deserializer.deserialize_with(&self.endorsement_serializer, input)
+                    let (rest, endo) = endorsement_deserializer
+                        .deserialize_with(&self.endorsement_serializer, input)?;
+
+                    if endo.verify_signature().is_err() {
+                        Err(nom::Err::Failure(ContextError::add_context(
+                            rest,
+                            "Endorsement signature invalid",
+                            ParseError::from_error_kind(rest, nom::error::ErrorKind::Fail),
+                        )))
+                    } else {
+                        Ok((rest, endo))
+                    }
```

Comment:
Would we prefer to add signature checking to the `deserialize_with` implementation?